src/nccl_ofi_rdma.c

```diff
@@ -6810,11 +6812,24 @@ static int ep_rail_init(nccl_net_ofi_rdma_ep_t *ep,
 	}
 #endif
 
+	if (tclass != FI_TC_UNSPEC) {
+		rail_info = fi_dupinfo(rail_info);
+		if (rail_info == NULL) {
+			NCCL_OFI_WARN("Could not allocate new fi_info struct");
+			return -ENOMEM;
+		}
+
```

Comment:
Why do we need a dup, can we not directly update dev_rail->info?

Reply:
`dev_rail->info` comes from the device and is reused every time an endpoint is created using that device, and is shared between control and data endpoints.  If we changed `dev_rail->info`'s TC, that would persist into the next data endpoint we created.  I suppose you could set it to `tclass` every time, but that would require a lot more thought about locking, and this is all in endpoint setup, so who cares about a host side copy?